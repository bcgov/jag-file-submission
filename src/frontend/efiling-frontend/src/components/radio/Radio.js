import React from "react";
import PropTypes from "prop-types";

import "./Radio.scss";

export const Radio = ({ id, name, label, onSelect, defaultChecked }) => (
  <label className="bcgov-radio" htmlFor={id}>
    {label}
    <input
      type="radio"
      name={name}
      id={id}
      onChange={(e) => onSelect(e.target.id)}
      defaultChecked={defaultChecked}
    />
    <span className="bcgov-dot" />
  </label>
);

Radio.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  onSelect: PropTypes.func,
  defaultChecked: PropTypes.bool,
};

Radio.defaultProps = {
  onSelect: () => {},
  defaultChecked: false,
};
