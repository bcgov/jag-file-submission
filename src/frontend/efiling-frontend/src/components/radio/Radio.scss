/* Customize the label (the container) */
.bcgov-radio {
  display: block;
  position: relative;
  padding-left: 30px;
  margin-bottom: 12px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

/* Hide the browser's default radio button */
.bcgov-radio input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

/* Create a custom radio button */
.bcgov-dot {
  position: absolute;
  top: 5px;
  left: 0;
  height: 18px;
  width: 18px;
  border-radius: 50%;
  border: 2px solid #606060;
}

/* When the radio button is checked, add a blue background */
.bcgov-radio input:checked ~ .bcgov-dot {
  background-color: #ffffff;
}

/* Custom checkbox has blue outline when in focus */
.bcgov-radio input:focus ~ .bcgov-dot {
  outline: 4px solid #3b99fc;
  outline-offset: 1px;
}

/* Create the indicator (the dot/circle - hidden when not checked) */
.bcgov-dot:after {
  content: "";
  position: absolute;
  display: none;
  top: 50%;
  left: 50%;
  width: 12px;
  height: 12px;
  border-radius: 50%;
  background: #606060;
  transform: translate(-50%, -50%);
}

/* Show the indicator (dot/circle) when checked */
.bcgov-radio input:checked ~ .bcgov-dot:after {
  display: block;
}
