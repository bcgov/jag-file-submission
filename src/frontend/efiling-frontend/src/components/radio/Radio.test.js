import React from "react";
import { render, fireEvent, getByRole } from "@testing-library/react";
import { Radio } from "./Radio";
import testBasicSnapshot from "../../TestHelper";

describe("Radio Component", () => {
  const id = "yes";
  const name = "yes";
  const label = "Yes";
  const onSelect = jest.fn();

  test("matches the snapshot", () => {
    testBasicSnapshot(
      <Radio id={id} name={name} label={label} onSelect={onSelect} />
    );
  });

  test("on change updates the selected value", () => {
    const { container } = render(
      <Radio id={id} name={name} label={label} onSelect={onSelect} />
    );

    fireEvent.click(getByRole(container, "radio"));

    expect(onSelect).toHaveBeenCalledWith(id);
  });
});
