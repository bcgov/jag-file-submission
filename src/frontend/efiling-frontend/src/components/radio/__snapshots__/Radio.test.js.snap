// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Component matches the snapshot 1`] = `
<label
  className="bcgov-radio"
  htmlFor="yes"
>
  Yes
  <input
    defaultChecked={false}
    id="yes"
    name="yes"
    onChange={[Function]}
    type="radio"
  />
  <span
    className="bcgov-dot"
  />
</label>
`;
